src/typingsInstallerCore/typingsInstaller.ts

```diff
@@ -31,28 +31,35 @@ namespace ts.server.typingsInstaller {
     }
 
     /*@internal*/
-    export function installNpmPackages(npmPath: string, tsVersion: string, packageNames: string[], install: (command: string) => boolean) {
+    export function installNpmPackages(npmPath: string, tsVersion: string, packageNames: string[], install: (file: string, args: string[]) => boolean) {
         let hasError = false;
         for (let remaining = packageNames.length; remaining > 0;) {
             const result = getNpmCommandForInstallation(npmPath, tsVersion, packageNames, remaining);
             remaining = result.remaining;
-            hasError = install(result.command) || hasError;
+            hasError = install(result.command[0], result.command[1]) || hasError;
         }
         return hasError;
     }
 
+    function getUserAgent(tsVersion: string) {
+        return `--user-agent="typesInstaller/${tsVersion}"`;
+    }
+    const npmInstall = "install", ignoreScripts = "--ignore-scripts", saveDev = "--save-dev";
+    const commandBaseLength = npmInstall.length + ignoreScripts.length + saveDev.length + getUserAgent("").length + 5;

```

Comment:
Should just say `getUserAgent("0.0.0")` rather than adding the `+ 5`, IMO.

Reply:
+ 5 is for the spaces around them (the tsversion length is added later on while computing actual command length as we use to.

Reply:
Ah. That was not immediately clear, sorry.